packages/stream/solana/StreamClient.ts

```diff
@@ -890,6 +894,28 @@ export default class SolanaStreamClient extends BaseStreamClient {
     return sortedStreams.filter((stream) => stream[1].type === type);
   }
 
+  public async searchStreams(data: ISearchStreams): Promise<IProgramAccount<Stream>[]> {
+    const filters: MemcmpFilter[] = [];
+    let k: keyof ISearchStreams;
+    for (k in data) {
+      const value = data[k];
+      if (value) {
+        filters.push({
+          memcmp: {
+            offset: STREAM_STRUCT_OFFSETS[k],
+            bytes: value,
+          },
+        });
+      }
+    }
```

Comment:
i'd say that we could make it simpler (but this is going to be my personal preference of doing things)

```suggestion
    const filters: MemcmpFilter[] = Object.entries(data)
      .filter(([_, value]) => value) // Only keep entries where the value is truthy
      .map(([key, value]) => ({
        memcmp: {
          offset: STREAM_STRUCT_OFFSETS[key as keyof ISearchStreams],
          bytes: value,
       },
    }));
```

or with `reduce` from Object.entries(data) syntax